app/src/main/java/com/andela/philskiiiwalker/levelup/view/MainActivity.java

```diff
@@ -1,29 +1,54 @@
 package com.andela.philskiiiwalker.levelup.view;
 
 import android.os.Bundle;
+import android.os.Parcelable;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
+import android.util.Log;
 
 import com.andela.philskiiiwalker.levelup.R;
+import com.andela.philskiiiwalker.levelup.adapter.GithubUsersAdapter;
+import com.andela.philskiiiwalker.levelup.model.GithubUsers;
 import com.andela.philskiiiwalker.levelup.presenter.GithubUsersPresenter;
 
+import java.util.ArrayList;
+import java.util.List;
 
-public class MainActivity extends AppCompatActivity {
-    private RecyclerView mRecyclerView;
-    private RecyclerView.LayoutManager mLayoutManager;
+
+public class MainActivity extends AppCompatActivity implements GithubUsersPresenter.View {
+    static final  String ALL_KEYS = "list_state";
+    ArrayList<GithubUsers> users;
+    private final GithubUsersPresenter presenter = new GithubUsersPresenter(this);
```

Comment:
@Kalela 
Instead of referencing the `GithubUsersPresenter` as the `type` of the `presenter` object, you can reference the `contract(interface)` instead.
For instance:
Instead of `GithubUsersPresenter presenter = new GithubUsersPresenter(this);`, you can write this `MainActivityContract.MainPresenter presenter = new GithubUsersPresenter(this);`